asset-transfer-basic/application-gateway-typescript/package.json

```diff
@@ -19,6 +19,7 @@
     "author": "Hyperledger",
     "license": "Apache-2.0",
     "dependencies": {
+        "@grpc/grpc-js": "^1.9.0",
```

Comment:
@bestbeforetoday Is this necessary?

Reply:
FYI i had to manually install it.

Reply:
Using `npm`, as the instructions for the samples do, this is not necessary. `@grpc/grpc-js` is a dependency of `@hyperledger/fabric-gateway` so gets installed and is visible to the sample application. `pnpm` looks to have been used instead of `npm`, and that behaves differently. `pnpm` only allows access to modules explicitly specified as dependencies in the application’s `package.json` file.

The reason the `@grpc/grpc-js`dependency was removed from the sample application’s `package.json` is that updates to `@hyperledger/fabric-gateway` meant that different dependency versions of `@grpc/grpc-js` could exist. This caused TypeScript errors when private properties of public gRPC classes changed. To minimise the chance of similar errors occurring in the future with the samples, the explicit dependency was removed.

Reply:
ok. Why is there no package-lock.json?

Reply:
What do you need the `package-lock.json` for?

Reply:
Ofcourse for dep version control

Reply:
I don’t really want to tie the sample applications to a specific patch release of `@hyperledger/fabric-gateway` and other Fabric packages. My preference would be for them to use the latest minor/patch release. Having a `package-lock.json` would require that to be updates for all samples on every release, which is something I would prefer not to have to do.

If the consensus is that having a `package-lock.json` tie dependencies down to specific versions that were used the last time the code was modified, I could live with that. It might mean issues like the gRPC library version incompatibility described above don’t surface so easily. You could argue this is good for users of the samples. You could argue this is worse for users of the Fabric APIs in general as issues are not identified and resolved so quickly.

In the past I have found occasionally instances where having a `package-lock.json` seems top have caused issues - perhaps with dependencies containing native code - when the runtime environment uses a different Node version or platform. I don’t know if this is something likely to cause issues here but it makes me wary of shipping `package-lock.json` files as part of samples.

Reply:
I thought these samples were for beginners and i wanted it to work as expected without any issues

Reply:
As I understand it, they do work as expected. The problem was that you used `pnpm` instead of `npm` as described in the instructions.

We could enhance the samples to ensure that they work with `pnpm` in addition to `npm`. I think this should include testing **all** of the samples using `pnpm` in the build, not just changing the dependencies for one sample.

We could also include `package-lock.json` files for **all** of the samples, not just one sample.

Those two changes are not related so _might_ be better tackled in separate pull requests.

Reply:
ok.